lib/openapi.js

```diff
@@ -0,0 +1,262 @@
+        ? transform(route.schema)
+        : route.schema
+
+      if (schema && schema.hide) {
+        continue
+      }
+
+      if (schema && schema.tags && schema.tags.includes(hiddenTag)) {
+        continue
+      }
+
+      const path = stripBasePath
+        ? stripBasePathByServers(route.url, swaggerObject.servers)
+        : route.url
+      const url = formatParamUrl(path)
+
+      const swaggerRoute = swaggerObject.paths[url] || {}
+
+      const swaggerMethod = {}
+      const parameters = []
+
+      // route.method should be either a String, like 'POST', or an Array of Strings, like ['POST','PUT','PATCH']
+      const methods = typeof route.method === 'string' ? [route.method] : route.method
+
+      for (const method of methods) {
+        swaggerRoute[method.toLowerCase()] = swaggerMethod
+      }
+
+      // All the data the user can give us, is via the schema object
+      if (schema) {
+        // the resulting schema will be in this order
+        if (schema.operationId) {
+          swaggerMethod.operationId = schema.operationId
+        }
+
+        if (schema.tags) {
+          swaggerMethod.tags = schema.tags
+        }
+
+        if (schema.summary) {
+          swaggerMethod.summary = schema.summary
+        }
+
+        if (schema.description) {
+          swaggerMethod.description = schema.description
+        }
+
+        if (schema.externalDocs) {
+          swaggerMethod.externalDocs = schema.externalDocs
+        }
+
+        if (schema.querystring) {
+          getQueryParams(parameters, schema.querystring)
+        }
+
+        if (schema.body) {
+          swaggerMethod.requestBody = {
+            content: {}
+          }
+          getBodyParams(swaggerMethod.requestBody.content, schema.body, schema.consumes)
+        }
+
+        if (schema.params) {
+          getPathParams(parameters, schema.params)
+        }
+
+        if (schema.headers) {
+          getHeaderParams(parameters, schema.headers)
+        }
+
+        if (parameters.length) {
+          swaggerMethod.parameters = parameters
+        }
+
+        if (schema.deprecated) {
+          swaggerMethod.deprecated = schema.deprecated
+        }
+
+        if (schema.security) {
+          swaggerMethod.security = schema.security
+        }
+
+        if (schema.servers) {
+          swaggerMethod.servers = schema.servers
+        }
+
+        for (const key of Object.keys(schema)) {
+          if (key.startsWith('x-')) {
+            swaggerMethod[key] = schema[key]
+          }
+        }
+      }
+
+      swaggerMethod.responses = genResponse(schema ? schema.response : null)
+
+      swaggerObject.paths[url] = swaggerRoute
+    }
+
+    if (opts && opts.yaml) {
+      const swaggerString = yaml.safeDump(swaggerObject, { skipInvalid: true })
+      cache.swaggerString = swaggerString
+      return swaggerString
+    }
+
+    cache.swaggerObject = swaggerObject
+    return swaggerObject
+
+    function getBodyParams (parameters, body, consumes) {
```

Comment:
Do we need to have all those functions defined in-line? Generally that causes issues for an optimizer.

Reply:
It use the swaggerObject inside, will passing a large object to everywhere have performance issue? If no, I will extract the function to other place.

Reply:
Since objects are passed by reference without copying in JavaScript, size is irrelevant, it should be very cheap.